contracts/Remittance.sol

```diff
@@ -1,16 +1,113 @@
 //SPDX-License-Identifier: MIT
 pragma solidity >= 0.6.0 < 0.7.0;
 
-import "../node_modules/@openzeppelin/contracts/access/Ownable.sol";
-import "../node_modules/@openzeppelin/contracts/utils/Pausable.sol";
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
 
-import {SafeMath} from "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeMath} from "./math/SafeMath.sol";
 
 contract Remittance is Ownable, Pausable {
 
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
+    mapping(address => uint) public balances;
 
+    mapping(bytes32 => RemittanceInstance) public remittances;
+
+    struct RemittanceInstance {
+        address funder;
+        address broker;
+        uint256 fundsOwed;
+
+    }
+
+    event Deposit(
+        address indexed depositor,
+        uint256 amount
+    );
+
+    event RemittanceCreated(
+        bytes32 indexed hash,
+        address indexed funder,
+        address indexed broker,
+        uint256 amount
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        address indexed broker,
+        uint256 amount
+    );
+
+    event WithDraw(
+        address indexed withdrawer,
+        uint256 amount
+    );
+
+    // Generic deposit function
+    function deposit() external payable whenNotPaused {
+        require(msg.value > 0, "The value must be greater than 0");
+        balances[msg.sender] = balances[msg.sender].add(msg.value);
+
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    // The funder can create a remittance
+    function create(bytes32 hash, address broker)
+    external
+    payable
+    whenNotPaused
+    returns(bool success)
+    {
+        emit RemittanceCreated(hash, msg.sender, broker, msg.value);
+        // check hash based on a bytes32 empty string or empty string hashed with sha3
+        require(hash != 0x00 && hash != 0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563, "Hash cannot be empty");
```

Comment:
I think you can make this magic number more explicit with a `bytes32 constant hash0 = keccak256(bytes32(0));`.

Reply:
Thank you. I have a much cleaner solution. Took me hours to figure out several permutations.